crates/sqlexec/src/remote/table.rs

```diff
@@ -86,10 +98,24 @@ impl TableProvider for StubRemoteTableProvider {
     async fn insert_into(
         &self,
         _state: &SessionState,
-        _input: Arc<dyn ExecutionPlan>,
+        input: Arc<dyn ExecutionPlan>,
     ) -> DfResult<Arc<dyn ExecutionPlan>> {
-        Err(DataFusionError::NotImplemented(
-            "insert_into called on a stub provider".to_string(),
-        ))
+        let broadcast_id = Uuid::new_v4();
+        let send = ClientExchangeSendExec {
+            broadcast_id,
+            client: self.client.clone(),
+            input,
+        };
+
+        let recv = RemoteExecutionExec::new(
+            self.client.clone(),
+            Arc::new(RemoteInsertExec::from_broadcast(
+                broadcast_id,
+                ProviderReference::RemoteReference(self.provider_id),
+            )),
+        );
+
+        let exec = SendRecvJoinExec::from_send_execs(Arc::new(recv), vec![send]);
+        Ok(Arc::new(exec))
```

Comment:
This should all be `Ok(Arc::new(RemoteInsertExec { provider: self.provider.clone(), input }))`.

Only place where we need to handle the client send/recv stuff is in the planner during rewriting.